src/chemistry/geoschem/cesmgc_history_mod.F90

```diff
@@ -0,0 +1,1191 @@
+
+       ! If wildcard is present
+       IF ( current%isWildcard ) THEN
+          ! Do nothing. This should never happen at this point since
+          ! Init_DiagList will exit with an error if wildcard is
+          ! encountered in HISTORY.rc while compiling with ESMF_.
+
+          ! When it comes time to implement, create exports in a loop,
+          ! either for all species or for advected species only. Include
+          ! a check that the export was not already created. Loop over
+          ! AdvNames if wildcard is ADV. Loop over SpecNames for all other
+          ! cases, passing not found = OK so that not all are necessarily
+          ! output. Later on, after species database is initialized, exports
+          ! for only species in the specific wildcard will be associated
+          ! with data and thus included in the output file.
+
+          ! If the meantime, skip wildcards if it gets here.
+          current => current%next
+          CYCLE
+       ENDIF
+
+       ! If this item is for a specific tag, append description.
+       ! This will need revisiting since there may be tag-dependent
+       ! strings to append to long names
+       IF ( current%isTagged ) THEN
+          desc = TRIM(desc) // " for " // TRIM(current%tag)
+       ENDIF
+
+       ! Create a new HistoryExportObj object
+       CALL Init_HistoryExport( am_I_Root, NewHistExp,         &
+                                name=current%name,             &
+                                metadataID=current%metadataID, &
+                                registryID=current%registryID, &
+                                long_name=desc,                &
+                                units=units,                   &
+                                vloc=vloc,                     &
+                                rank=rank,                     &
+                                type=type,                     &
+                                isMet=isMet,                   &
+                                isChem=isChem,                 &
+                                isDiag=isDiag,                 &
+                                RC=RC )
+       IF ( RC == GC_FAILURE ) THEN
+          RC = GC_FAILURE
+          ErrMsg = "History export init fail for " // TRIM(current%name)
+          EXIT
+       ENDIF
+
+       ! Add new HistoryExportObj to linked list
+       CALL Append_HistoryExportsList( am_I_Root,     NewHistExp, &
+                                       HistoryConfig, RC       )
+       IF ( RC == GC_FAILURE ) THEN
+          RC = GC_FAILURE
+          ErrMsg = "History export append fail for " // TRIM(current%name)
+          EXIT
+       ENDIF
+
+       ! Set up for next item in DiagList
+       current => current%next
+
+    ENDDO
+    current => NULL()
+
+    IF ( RC == GC_SUCCESS ) THEN
+       HistoryConfig%ConfigFileRead = .TRUE.
+    ELSE
+       CALL GC_ERROR( ErrMsg, RC, Iam )
+       _ASSERT(.FALSE., 'informative message here')
```

Comment:
I know you added the _ASSERT macro locally. Out of curiosity, does CESM have something that is equivalent for this functionality? The use case here is a bit weird. Usually it's something like _ASSERT(A>0,'Error...')

Reply:
I don't think there is - it's usually just IF calls and call to `endrun`.